src/model/package.py

```diff
@@ -0,0 +1,144 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from datetime import datetime
+from enum import Enum, auto
+from typing import Optional, Union, List
+
+from src.model.actor import Actor
+from src.model.checksum import Checksum
+from src.model.license_expression import LicenseExpression
+from src.model.spdx_no_assert import SpdxNoAssert
+from src.model.spdx_none import SpdxNone
+
+
+class PackagePurpose(Enum):
+    APPLICATION = auto()
+    FRAMEWORK = auto()
+    LIBRARY = auto()
+    CONTAINER = auto()
+    OPERATING_SYSTEM = auto()
+    DEVICE = auto()
+    FIRMWARE = auto()
+    SOURCE = auto()
+    ARCHIVE = auto()
+    FILE = auto()
+    INSTALL = auto()
+    OTHER = auto()
+
+
+class PackageVerificationCode:
+    value: str
+    excluded_files: List[str]
+
+    def __init__(self, value: str, excluded_files: List[str] = None):
+        self.value = value
+        self.excluded_files = [] if excluded_files is None else excluded_files
+
+
+class ExternalPackageReferenceCategory(Enum):
+    SECURITY = auto()
+    PACKAGE_MANAGER = auto()
+    PERSISTENT_ID = auto()
+    OTHER = auto()
+
+
+class ExternalPackageReference:
+    category: ExternalPackageReferenceCategory
+    # In theory, once could refine the typing,
+    # see https://spdx.github.io/spdx-spec/v2.3/external-repository-identifiers/. But it's probably not worth the
+    # effort.
+    reference_type: str
+    locator: str
+    comment: Optional[str]
+
+    def __init__(self, category: ExternalPackageReferenceCategory, reference_type: str, locator: str,
+                 comment: Optional[str] = None):
+        self.category = category
+        self.reference_type = reference_type
+        self.locator = locator
+        self.comment = comment
+
+
+class Package:
+    # Mandatory properties
+
+    spdx_id: str
+    name: str
+    download_location: Union[str, SpdxNoAssert, SpdxNone]
+
+    # Optional properties
+
+    version: Optional[str]
+    file_name: Optional[str]
+    supplier: Optional[Actor, SpdxNoAssert]
+    originator: Optional[Actor, SpdxNoAssert]
+    files_analyzed: bool  # defaults to True
+    verification_code: Optional[PackageVerificationCode]
+    checksums: List[Checksum]
+    homepage: Optional[str, SpdxNoAssert, SpdxNone]
+    source_info: Optional[str]
+    concluded_license: Optional[LicenseExpression, SpdxNoAssert, SpdxNone]
+    license_info_from_files: Optional[List[LicenseExpression], SpdxNoAssert, SpdxNone]
+    license_declared: Optional[LicenseExpression, SpdxNoAssert, SpdxNone]
```

Comment:
```suggestion
    declared_license: Optional[LicenseExpression, SpdxNoAssert, SpdxNone]
```

Reply:
The property names in the spec are always "[...] license declared": https://spdx.github.io/spdx-spec/v2.3/package-information/#715-declared-license-field

Reply:
Well ok, then we should rename `concluded_license`, though.

Reply:
Fine, you win :sweat_smile:

Reply:
Well, actually you had me convinced to use the `license_concluded` so as not to get confused when writer and parser use this ordering :D